packages/cli/src/commands/versions/bump.ts

```diff
@@ -223,12 +233,27 @@ export default async (opts: OptionValues) => {
           for (const depType of DEP_TYPES) {
             if (depType in pkgJson && dep.name in pkgJson[depType]) {
               const oldRange = pkgJson[depType][dep.name];
-              pkgJson[depType][dep.name] = dep.range;
+
+              // backstage:^ are written to the lockfile as
+              // backstage:<backstage-version>, so that updates to
+              // backstage.json can be detected during yarn install. In order to
+              // locate the corresponding lockfile entry for "backstage:^"
+              // versions, we need to perform the same transformation.
+              const oldLockfileRange = await asLockfileVersion(oldRange);
+
+              // Don't use backstage:^ versions for peerDependencies; they only
+              // support npm and workspace: versions.
+              const useBackstageRange =
+                hasYarnPlugin && depType !== 'peerDependencies';
+
+              const newRange = useBackstageRange ? 'backstage:^' : dep.range;
+
+              pkgJson[depType][dep.name] = newRange;
```

Comment:
This seems a bit aggressive, we probably don't want this to happen unless the version manifest we're bumping to also contains the package?

Reply:
Yep, makes sense! I guess this mostly covers downgrades, and packages moving out of the @backstage scope and into the community repo.

Reply:
Done in 9def475ed084.